src/utils/config.ts

```diff
@@ -30,9 +30,17 @@ export const IConfig = t.interface({
 // No need to re-evaluate this object for each call
 const errorOrConfig: t.Validation<IConfig> = IConfig.decode({
   ...process.env,
-  SERVER_PORT: fromNullableE(-1)(process.env.SERVER_PORT)
-    .chain(_ => IntegerFromString.decode(_).mapLeft(() => -1))
-    .fold(identity, identity),
+  SERVER_PORT: pipe(
+    process.env.SERVER_PORT,
+    E.fromNullable(-1),
+    E.chain(
+      flow(
+        IntegerFromString.decode,
+        E.mapLeft(() => -1)
+      )
+    ),
+    E.toUnion
+  ),
```

Comment:
Maybe it can be simplified:

```suggestion
  SERVER_PORT: pipe(
    process.env.SERVER_PORT,
    IntegerFromString.decode,
    E.getOrElse(() => -1)
,
```